displayChallenges.js

```diff
@@ -0,0 +1,282 @@
+const topThreeInIndex = document.querySelectorAll("section.challenges > ul > li");
+
+async function threeHighestRanked() {
+    let topThreeByRating = [];
+    const res = await fetch('https://lernia-sjj-assignments.vercel.app/api/challenges');
+    const data = await res.json();
+    data.challenges.forEach(challenge => {
+        if (challenge.rating > 4 && topThreeByRating.length < 3) {
+            topThreeByRating.push(challenge);
+        }
+    })
+    if (topThreeByRating.length < 3) {
+        data.challenges.forEach(challenge => {
+            if (challenge.rating > 3 && challenge.rating <= 4) {
+                topThreeByRating.push(challenge);
+            }
+        })
+    }
```

Comment:
I see what you've attempted to do here, but the logic is flawed. What happens if the highest rating is 3?

This is the biggest challenge every programmer faces – how to design algorithms. Learning how to write code – the syntax and so on – is usually done in a few months, but how to solve problems with code is a life-long learning process.

Try to think about how you would solve this in real life, e.g. if I gave you 15 random cards from a deck of cards, and asked you to give me the five highest ones, how would you do it as a human? How does that translate to how a computer could do it? That's how you could approach problems like this.